src/Microsoft.Identity.Web/ClaimsPrincipalFactory.cs

```diff
@@ -107,5 +107,36 @@ public static ClaimsPrincipal FromTenantIdAndObjectId(string tenantId, string ob
                     }));
             }
         }
+
+        /// <summary>
+        /// Instantiate a <see cref="ClaimsPrincipal"/> from a username and password.
+        /// This can be used for ROPC flow for testing purposes.
+        /// </summary>
+        /// <param name="username">UPN of the user for example username@domain.</param>
+        /// <param name="password">Password for the user.</param>
+        /// <returns>A <see cref="ClaimsPrincipal"/> containing these two claims.</returns>
+        public static ClaimsPrincipal FromUsernamePassword(string username, string password)
+        {
+            if (AppContextSwitches.UseClaimsIdentityType)

```

Comment:
This if () is not needed.
this is a new API. Just use `CaseSensitiveClaimsIdentity`